lib/database/repositories/QcFlagRepository.js

```diff
@@ -123,6 +134,118 @@ class QcFlagRepository extends Repository {
+                ON qcf.id = qcfep.flag_id
+            INNER JOIN data_pass_quality_control_flag AS dpqcf
+                ON dpqcf.quality_control_flag_id = qcf.id
+            INNER JOIN (${GAQ_PERIODS_VIEW}) AS gaq_periods
+                ON gaq_periods.data_pass_id = dpqcf.data_pass_id
+            INNER JOIN global_aggregated_quality_detectors AS gaqd
+                ON gaqd.data_pass_id = gaq_periods.data_pass_id
+                    AND gaqd.run_number = gaq_periods.run_number
+                    AND gaqd.detector_id = qcf.detector_id
+                    AND gaq_periods.run_number = qcf.run_number
+                        AND (qcfep.\`from\` IS NULL OR UNIX_TIMESTAMP(qcfep.\`from\`) <= gaq_periods.\`from\`)
+                        AND (qcfep.\`to\`   IS NULL OR gaq_periods.\`to\` <= UNIX_TIMESTAMP(qcfep.\`to\`))
+
+            GROUP BY
+                gaq_periods.data_pass_id,
+                gaq_periods.run_number,
+                gaq_periods.\`from\`,
+                gaq_periods.\`to\`
+            `;
+
+        const query = `
+            SELECT
+                effectivePeriods.runNumber,
+                effectivePeriods.dataPassId,
+                effectivePeriods.bad,
+                SUM(effectivePeriods.mcReproducible) > 0 AS mcReproducible,
+                GROUP_CONCAT(effectivePeriods.verifiedFlagsList) AS verifiedFlagsList,
+                GROUP_CONCAT(effectivePeriods.flagsList) AS flagsList,
+
+                IF(
+                    (
+                        COALESCE(run.time_trg_end,   run.time_o2_end  ) IS NULL
+                     OR COALESCE(run.time_trg_start, run.time_o2_start) IS NULL
+                    ),
+                    IF(
+                        SUM(
+                            COALESCE(effectivePeriods.\`to\`  , 0)
+                          + COALESCE(effectivePeriods.\`from\`, 0)
+                        ) = 0,
+                        1,
+                        null
+                    ),
+                    SUM(
+                        COALESCE(
+                            effectivePeriods.\`to\`,
+                            UNIX_TIMESTAMP(run.time_trg_end),
+                            UNIX_TIMESTAMP(run.time_o2_end)
+                        )
+                      - COALESCE(
+                            effectivePeriods.\`from\`,
+                            UNIX_TIMESTAMP(run.time_trg_start),
+                            UNIX_TIMESTAMP(run.time_o2_start)
+                        )
+                    ) / (
+                        UNIX_TIMESTAMP(COALESCE(run.time_trg_end,   run.time_o2_end))
+                      - UNIX_TIMESTAMP(COALESCE(run.time_trg_start, run.time_o2_start))
+                    )
+                ) AS effectiveRunCoverage
+
+            FROM (${effectivePeriodsWithTypeSubQuery}) AS effectivePeriods
+            INNER JOIN runs AS run ON run.run_number = effectivePeriods.runNumber
+
+            WHERE effectivePeriods.dataPassId = ${dataPassId}
```

Comment:
Same as previously, but here it is more important. `dataPassId` comes from user request. It should be using `sequelize.replacements` rather than injecting directly in the query

Reply:
applied